src/libServer/Server.cpp

```diff
@@ -267,6 +300,21 @@ Json::Value Server::CreateTransaction(const Json::Value& _json) {
   }
 }
 
+Server::ContractType Server::GetTransactionType(const Transaction& tx) const {
+  if (tx.GetData().empty() || tx.GetToAddr() == NullAddress) {
+    if (tx.GetData().empty() && tx.GetCode().empty() &&
+        tx.GetToAddr() != NullAddress) {
+      return NON_CONTRACT;
+    } else if (!tx.GetCode().empty() && tx.GetToAddr() == NullAddress) {
+      return CONTRACT_CREATION;
+    } else {
+      return ERROR;
+    }
+  } else {
+    return CONTRACT_CALL;
+  }
+}
+
```

Comment:
How about this

```
  if (!tx.GetData().empty() && tx.GetToAddr() != NullAddress) { 
    return CONTRACT_CALL;
  }

  if (!tx.GetCode().empty() && tx.GetToAddr() == NullAddress) {
    return CONTRACT_CREATION;
  }

  if (tx.GetData().empty() && tx.GetToAddr() != NullAddress && tx.Getcode().emtpy()) {
    return NON_CONTRACT;
  }

  return ERROR;
```

Need someone else to confirm this is equivalent

Reply:
Your version looks equivalent to me.

Reply:
@Gnnng looks equivalent to me too. I have pushed the change